src/components/MediaCreationModal/MediaCreationModal.tsx

```diff
@@ -0,0 +1,350 @@
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  useDisclosure,
+  Text,
+  Icon,
+  useModalContext,
+  Progress,
+  ListItem,
+  UnorderedList,
+  VStack,
+  Flex,
+} from "@chakra-ui/react"
+import { Button, Infobox } from "@opengovsg/design-system-react"
+import { useEffect, useState } from "react"
+import { FileRejection } from "react-dropzone"
+import { BiCheckCircle, BiSolidErrorCircle } from "react-icons/bi"
+import { useParams } from "react-router-dom"
+
+import { Attachment } from "components/Attachment"
+
+import { useCreateMultipleMedia } from "hooks/mediaHooks/useCreateMultipleMedia"
+
+import { MediaDirectoryParams } from "types/folders"
+
+import { Dropzone } from "./components/Dropzone"
+
+type MediaSteps = "upload" | "progressing" | "success" | "failed"
+type UploadVariant = "files" | "images"
+
+const IMAGE_UPLOAD_ACCEPTED_MIME_TYPES = {
+  "image/jpeg": [".jpg", ".jpeg"],
+  "image/png": [".png"],
+  "image/gif": [".gif"],
+  "image/svg+xml": [".svg"],
+  "image/tiff": [".tiff", ".tif"],
+  "image/bmp": [".bmp"],
+}
+
+const FILE_UPLOAD_ACCEPTED_MIME_TYPES = {
+  "application/pdf": [".pdf"],
+}
+
+interface MediaDropzoneProps {
+  fileRejections: FileRejection[]
+  uploadedFiles: File[]
+  setUploadedFiles: (files: File[]) => void
+  setFileRejections: (rejections: FileRejection[]) => void
+  isDisabled?: boolean
+  onUpload: (files: File[]) => Promise<void>
+  variant: UploadVariant
+}
+
+const MediaDropzone = ({
+  fileRejections,
+  uploadedFiles,
+  setUploadedFiles,
+  setFileRejections,
+  isDisabled,
+  onUpload,
+  variant,
+}: MediaDropzoneProps) => {
+  const { onClose } = useModalContext()
+
+  return (
+    <>
+      <ModalHeader>Upload files</ModalHeader>
+      <ModalBody>
+        <Text textStyle="body-1" mb="1.5rem">
+          You can upload more than 1 file at once. Having too many files can
+          slow down the site loading time, so we recommend only uploading
+          necessary files to your site.
+        </Text>
+        <Attachment
+          rejected={fileRejections}
+          accept={
+            variant === "files"
+              ? FILE_UPLOAD_ACCEPTED_MIME_TYPES
+              : IMAGE_UPLOAD_ACCEPTED_MIME_TYPES
+          }
+          onChange={(curUploadedFiles, rejections) => {
+            setUploadedFiles([...uploadedFiles, ...curUploadedFiles])
+            setFileRejections([...fileRejections, ...rejections])
+          }}
+          value={uploadedFiles}
+          name=""
+          // NOTE: 5MB - maxSize is in bytes
+          maxSize={5 * 10 ** 6}
```

Comment:
nit: we should store the `5` value as a constant somewhere else (seen this being used multiple times)

Reply:
can, will update